string piece_color
string category
string square_color
